rllib/core/models/tf/base.py

```diff
@@ -0,0 +1,65 @@
+import abc
+from typing import Tuple
+
+import numpy as np
+
+from ray.rllib.core.models.base import (
+    Model,
+    ModelConfig,
+    _raise_not_decorated_exception,
+)
+from ray.rllib.models.specs.checker import (
+    check_input_specs,
+    check_output_specs,
+)
+from ray.rllib.models.specs.checker import (
+    is_input_decorated,
+    is_output_decorated,
+)
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.nested_dict import NestedDict
+
+_, tf, _ = try_import_tf()
+
+
+class TfModel(Model, tf.keras.Model, abc.ABC):
+    """Base class for RLlib's TensorFlow models.
+
+    This class defines the interface for RLlib's TensorFlow models and checks
+    whether inputs and outputs of __call__ are checked with `check_input_specs()` and
+    `check_output_specs()` respectively.
+    """
+
+    def __init__(self, config: ModelConfig):
+        tf.keras.Model.__init__(self)
+        Model.__init__(self, config)
+
+        # Raise errors if forward method is not decorated to check specs.
+        if not is_input_decorated(self.__call__):
```

Comment:
wait `self.call`, right? `__call__` won't be decorated, right?

Reply:
Great catch man! Fixed.